PayloadPkg/FirmwareUpdate/FirmwareUpdateHelper.c

```diff
@@ -736,14 +739,28 @@ UpdateContainerComp (
   // Update the component
   //
   ContainerHdr = (CONTAINER_HDR *)(UINTN)ContainerEntryPtr->HeaderCache;
-  //
-  // Component base = Container base + data offset from container base + offset of component inside container
-  //
-  ComponentBase = ContainerEntryPtr->Base + ContainerHdr->DataOffset + ComponentEntryPtr->Offset;
+
+  if (ContainerEntryPtr->Base >= 0xF0000000) {
```

Comment:
As a future enhancement, we could get this MMIO flash limit more dynamically and avoid hard coding it.

Reply:
0xF0000000 (256MB flash space) should be safe. RAM memory will not use this range.